Make the compile work on a Mac

ballanty opened the pull request:
This change:

- sed as used required the addition of a small change to have it work on the Mac. The Makefile was altered to detect the platform and conditionally use sed in different ways in one case.
- it looks like "packets" was once a buffer size measure and it was changed to bytes at some point.  Fix what appears to be a leftover reference to packets.
- parsing of quantities ("2days" for example) seemed to differ by platform (Mac (didn't work) vs Linux (did)).  Replaced parsing approach with one using regular expressions to make it more robust to platform differences between the way std::..stream operator>> parsed.

ballanty committed:
Make it run on a mac

ballanty committed:
Change Makefile to use sed hack

ballanty committed:
Small adjustment

ballanty committed:
Another Makefile adjustment

ballanty committed:
Re-add the Optimization flag to the compiler flags.

lorenzosaino commented on src/quantity.cpp:
```diff
@@ -20,8 +21,19 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
-	std::istringstream ss(str);
-	ss>>this->value>>this->unit;
+
+    std::string regexpstr = "([0-9]+)(\\.[0-9]*)? *([^ \t]*)";
+    std::regex reg(regexpstr);
+    std::smatch match;
+    std::regex_search(str, match, reg);
+
+    if ( !match.empty() ) {
+        this->value = std::stod(match[1].str() + match[2].str());
+        this->unit = match[3];
+    } else {
+        this->value = 0.0;
+        this->unit = "";
+    }
```
Minor nit: this block uses spaces for indentation, while the rest of the code uses tabs. I'd suggest we use tabs here too for consistency

lorenzosaino commented on src/quantity.cpp:
```diff
@@ -20,8 +21,19 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
+    std::smatch match;
+    std::regex_search(str, match, reg);
+
+    if ( !match.empty() ) {
```
Minor nit: in the rest of the code, there are not spaces between parentheses and conditions (see for example at L38 below). I would use the same convention for consistency

lorenzosaino commented on test/test.cpp:
```diff
@@ -520,11 +522,11 @@ void testEvent() {
 	Quantity t3("60min", Units::Time);
 	Quantity t4("3601 sec", Units::Time);
-	t1.fromString("2days");
+	//t1.fromString("2days");
```
Is commenting this out intended? Without this we are no longer testing anywhere the `fromString` method

lorenzosaino commented on Makefile:
```diff
@@ -159,8 +167,8 @@ run_test    = printf "Executing:   $1:  "; \
              else \
-                 rm $1; \
-                 exit 1; \
+                  rm $1; \
+                  exit 1; \
```
Minor nit: not sure if intended but by adding space in front of these two lines makes them no longer aligned with the `echo "OK"; \ ` instruction above. Was this change intended?

lorenzosaino commented on test/test.cpp:
```diff
@@ -302,6 +302,8 @@ void testEdge() {
 
 	Edge e;
+
+    e.setBufferSize(Quantity("1kB", Units::Data));
```
Minor nit like my comment above about using tabs instead of spaces

lorenzosaino reviewed (changes requested):
Thanks for contributing this feature. I made a few comments about minor nits, but the core change looks good.

ballanty committed:
Changes post review comments:
- fixed spaces to tabs in a few spots (tabs project default)
- fixed spacing conventions around if conditions.
- fixed spaces in shell script - not the same length as surrounding code
- reenabled a test after I commented it out to get changes working

ballanty commented:
Good catches all.  I'm sorry I made you go through the detailed review.  I'll remember the project conventions going forward!

lorenzosaino reviewed (approved):
Looks great, thank you!
